regressiontests/regression_test.go

```diff
@@ -0,0 +1,391 @@
+			if err != nil {
+				t.Fatalf("Couldn't prepare HTTP request %v", err)
+			}
+
+			resp, err := client.Do(req)
+			if err != nil {
+				t.Logf("Failed to make HTTP request to the application: %v - next attempt in 10s", err)
+				continue
+			}
+
+			if closeErr := resp.Body.Close(); closeErr != nil {
+				t.Errorf("failed to close response body: %v", closeErr)
+			}
+
+			if resp.StatusCode != 200 {
+				t.Logf("Expected status code 200, got %d - next attempt in 10s", resp.StatusCode)
+				continue
+			}
+
+			t.Log("HTTP server is up and running")
+
+			wocTipHeight, err := fetchExternalForkHeight(ctx, whatsonchainAPIURL, t)
+			if err != nil {
+				t.Logf("Failed to fetch external fork height: %v", err)
+				continue
+			}
+
+			localTipHeight, err := fetchLocalTip(ctx, t)
+			if err != nil {
+				t.Logf("Failed to fetch local tip height: %v", err)
+				continue
+			}
+
+			if localTipHeight < wocTipHeight {
+				t.Logf("Couldn't sync to proper tip of chain - next attempt in 10s: %d < %d", localTipHeight, wocTipHeight)
+				continue
+			} else {
+				t.Logf("Synced to tip of chain: %d", localTipHeight)
+				break out
+			}
+		}
+	}
+
+	t.Log("Comparing synced data with known checkpoints...")
+
+	verifyHeaders(ctx, t)
+
+	t.Log("Headers checkpoint comparison passed successfully 🎉")
+
+	t.Log("Verifying merkle roots...")
+
+	verifyMerkleRoots(ctx, t, fixtures)
+
+	t.Log("Merkle roots verification passed successfully 🎉")
+}
+
+func setPostgresEnvs(cmd *exec.Cmd, mappedPort string) {
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "BHS_DB_POSTGRES_PORT="+mappedPort)
+	cmd.Env = append(cmd.Env, "BHS_DB_ENGINE=postgres")
+	cmd.Env = append(cmd.Env, "BHS_DB_PREPARED_DB=true")
+	cmd.Env = append(cmd.Env, "BHS_DB_PREPARED_DB_FILE_PATH=../data/blockheaders.csv.gz")
+	cmd.Env = append(cmd.Env, "BHS_DB_SCHEMA_PATH=../database/migrations")
+}
+
+func setSQLiteEnvs(cmd *exec.Cmd) {
+	cmd.Env = append(cmd.Env, "BHS_DB_ENGINE=sqlite")
+	cmd.Env = append(cmd.Env, "BHS_DB_PREPARED_DB=true")
+	cmd.Env = append(cmd.Env, "BHS_DB_PREPARED_DB_FILE_PATH=../data/blockheaders.csv.gz")
+	cmd.Env = append(cmd.Env, "BHS_DB_SCHEMA_PATH=../database/migrations")
+	cmd.Env = append(cmd.Env, "BHS_DB_SQLITE_FILE_PATH=../data/blockheaders.db")
+}
+
+func startPostgresContainer(ctx context.Context, t *testing.T) (testcontainers.Container, string) {
+	req := testcontainers.ContainerRequest{
+		Image:        "postgres:latest",
+		ExposedPorts: []string{"5432/tcp"},
+		Env: map[string]string{
+			"POSTGRES_USER":     "user",
+			"POSTGRES_PASSWORD": "password",
+			"POSTGRES_DB":       "bhs",
+		},
+		WaitingFor: wait.ForListeningPort("5432/tcp"),
+	}
+	postgresContainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		t.Fatalf("Could not start postgres container: %v", err)
+	}
+
+	mappedPort, err := postgresContainer.MappedPort(ctx, "5432")
```

Comment:
AFAIU it will map the container port to some random port and this is the way how we're getting this random port. 
Awesome!